### ISSUE TITLE
Challenge 1

### ISSUE THREAD

**Issue by 0xStryK3R** [scaleracademy/scaler-september-open-source-challenge#136]
name: Arup Bhowmick
github_user_name: 0xStryK3R

**Comment by 0xStryK3R** [comment]
- [x] URL of your forked repository [https://github.com/0xStryK3R/scaler-september-open-source-challenge](https://github.com/0xStryK3R/scaler-september-open-source-challenge)
- [x] A screenshot of the cloned project in your system:

![image](https://user-images.githubusercontent.com/29790345/188118027-650a0d5e-7a1b-48f1-b028-d765963573d0.png)

**Comment by 0xStryK3R** [comment]
- [x] Screenshot showing the list of all the current branches inside of your forked clone repository

![image](https://user-images.githubusercontent.com/29790345/188259745-e8996f83-334d-481b-ad39-c231b7789fe1.png)

**Comment by 0xStryK3R** [comment]
- [x] Screenshot showing the latest repository contents

- [x] Main Branch Updated
![image](https://user-images.githubusercontent.com/29790345/188308832-01b510d1-321b-477b-8510-6f19cade9990.png)

- [x] Personal Branch Updated
![image](https://user-images.githubusercontent.com/29790345/188308841-97ae22b6-129e-4823-88f5-5283be5f3ed5.png)

- [x] Local Repo Updated
![image](https://user-images.githubusercontent.com/29790345/188308746-e7cee63b-a77c-47f7-bf2e-acd1cc68b62f.png)

**Comment by 0xStryK3R** [comment]
## Challenge 9 Tasks:

 - [x] Identify the commit id that needs to be removed (you should have created this in the 8th Challenge)
![image](https://user-images.githubusercontent.com/29790345/189308127-0b01c4e2-51a4-4e80-9bf8-8ba760ff7057.png)

 - [x] Use git reset and force push, to remove the commit from your pull request. 
![image](https://user-images.githubusercontent.com/29790345/189308367-234ca55e-d371-488e-b047-5ff059084266.png)

 - [x] Share a screenshot of this, in the issue you created in previous challenges. #902 
![image](https://user-images.githubusercontent.com/29790345/189308555-9720cbb3-ac8c-4b30-b3ac-edc35d2ca17a.png)

**Comment by 0xStryK3R** [comment]
- [x] Challenge 10 Task Completed
- [x] Milestone 1 (Day 10) Reached
- [x] [Twitter Post Link](https://twitter.com/0xStryK3R/status/1568583047712505859?t=6tDUjQNL21r5hXBqT_unwg&s=19)

**Comment by 0xStryK3R** [comment]
Challenge 11 Tasks:

- [x] Create another branch from your current branch created in previous challenges (Don't create a new branch from main).
- [x] Share screenshot of this new branch in the issue created in previous challenge

![image](https://user-images.githubusercontent.com/29790345/189531032-ff450aef-a66d-4dcf-9d0e-b66d41c57f03.png)

**Comment by 0xStryK3R** [comment]
Challenge 13 Tasks:

- [x] Inside the branch you created in the previous challenge your-username-details create a new file inside the contributors/<YOUR-GITHUB_USERNAME>/ directory.
- [x] Use the git stash command to move these changes to the stash area
- [x] Delete the content from the stash area and share a screenshot of this inside the Github issue you created in the first challenge

![image](https://user-images.githubusercontent.com/29790345/189871324-7cac7dec-eaf7-46f2-8200-28dd29244ccc.png)

**Comment by 0xStryK3R** [comment]
Challenge 14 Tasks:

- [x] Follow the article shared above and try to reproduce a case of merge conflict in your system for a project and try to resolve it
- [x] Share appropriate screenshot in the issue created in the first challenge.

![image](https://user-images.githubusercontent.com/29790345/190140019-ff156500-6618-4296-97c9-fd4143c2e38f.png)

![image](https://user-images.githubusercontent.com/29790345/190140590-b409bdc0-a358-4e3c-ad94-38f4b179037d.png)

![image](https://user-images.githubusercontent.com/29790345/190141782-ec870c04-224b-4f59-abe2-234ff4b274fe.png)

**Comment by DaltonDeven** [comment]
Good to go verified ✅

**Comment by 0xStryK3R** [comment]
### Challenge 15 Tasks:

- [x] Update your Forked Repo From the Original Repo. Also ensure that your branch your-user-name-details is also synced with the latest changes from the original repository

- [x] In your branch your-user-name-details open the add your name to the list of challengers inside the challengers-list.md file by following the format in the file.

- [x] Push your change to reflect in the Pull Request created in previous challenges. 

- #902 

![image](https://user-images.githubusercontent.com/29790345/190404914-517fee8f-aa97-453b-98e4-ed9095b58f53.png)

**Comment by 0xStryK3R** [comment]
### Challenge 16 Tasks:

- [x] You will notice that your pull request will have a conflict: image

- [x] Try to resolve this merge conflict using local git

- [x] Once resolved, share screenshot on the github issue created in the previous challenges

- #902 

![image](https://user-images.githubusercontent.com/29790345/190632121-5335c9f7-ec07-4b81-bf7a-be8a88134585.png)

![image](https://user-images.githubusercontent.com/29790345/190632953-1d4de461-adf3-49d6-98fd-aba3092105f6.png)

![image](https://user-images.githubusercontent.com/29790345/190633641-f351520a-9b17-492f-9883-abd53a1edb9f.png)

![image](https://user-images.githubusercontent.com/29790345/190633737-01e1a636-fb57-4b67-8698-aa0915cffcf8.png)

**Comment by 0xStryK3R** [comment]
### Challenge 17 Tasks:

- [x] Download GitHub desktop
![image](https://user-images.githubusercontent.com/29790345/190858534-8fab880a-0e34-4fbf-a776-a2f054be2895.png)

- [x] Create a new repository on GitHub
![image](https://user-images.githubusercontent.com/29790345/190858738-3cca6347-255b-4ff0-a6e1-ef3bd3c69cfd.png)

- [x] Add anything you want to your repo
![image](https://user-images.githubusercontent.com/29790345/190858968-1ee1ef9f-248a-4c4e-9834-eb3fc905f955.png)

- [x] Follow through all the GitHub workflow you know using the GUI (commit, branching, pulling, pushing etc)
    - [x] Publish Repo to Github [Repo Link](https://github.com/0xStryK3R/oss-challenge-17)
    ![image](https://user-images.githubusercontent.com/29790345/190859026-da674f5c-ea3e-4ca9-8062-906828b28857.png)
    - [x] Commit Changes to README.md
    ![image](https://user-images.githubusercontent.com/29790345/190859127-244792fe-e324-4913-b41f-64dc4a2db543.png)
    - [x] Push `main` branch to Origin
    ![image](https://user-images.githubusercontent.com/29790345/190859274-dd0ff21b-0d6a-42be-b058-4a2ba7a013b7.png)
    - [x] Create New Branch (`test-branch`)
    ![image](https://user-images.githubusercontent.com/29790345/190859165-0a883ec2-f2d6-4051-bb2f-2226be544f53.png)
    - [x] Committed New Changes from `test-branch`
    ![image](https://user-images.githubusercontent.com/29790345/190859482-c7589237-f4af-4db6-8672-0c26f987a003.png)
    - [x] Published `test-branch`
    ![image](https://user-images.githubusercontent.com/29790345/190859531-f6b8104b-1271-4631-b4fb-5a820e25ecac.png)

- [x] Share a screenshot of your experience using github desktop on the issue created in prior challenges
![image](https://user-images.githubusercontent.com/29790345/190859596-059e56c4-d5c6-4bd2-8371-b5ef4f601c61.png)

**Comment by 0xStryK3R** [comment]
### Challenge 19 Tasks:

- [x] Create a new file inside your branch created in previous challenges your-username-details under the directory contributors/username. The file needs to be a source code file written in your preferred programming language. The code can be anything you want: hello world program or your favorite program.
- [x] Commit the change and push it.
- [x] Check if your linting fails in side the Pull request: `passed`

- #902 

![image](https://user-images.githubusercontent.com/29790345/190985831-47a11c03-981c-48cf-bb3b-8005d6753f65.png)

**Comment by 0xStryK3R** [comment]
### Challenge 20 Tasks:

- [x] Create a GitHub repository for any project you can think up: [Repo Link](https://github.com/0xStryK3R/url-shortener)
- [x] On a sheet of paper create a TODO list (or backlog) of features you want to have in you app
- [x] Create a project board with a TODO column, In progress, Done [Project Board](https://github.com/users/0xStryK3R/projects/4)
- [x] Create a card for all the items you already wrote on the sheet of paper in the TODO column
- [x] Work on the features synchronously and move the card respectively for each in progress task to the In progress column
- [x] When you're done with the In progress task move the card to the Done column

## Submission:
- [x] Take a screenshot of your project board for when you moved an card from In progress to Done
- [x] Add you screenshot to the Github issue created in the previous challenges


### Board View:
![image](https://user-images.githubusercontent.com/29790345/191282954-989e0408-f4af-4c2c-b6a4-e637187ca683.png)
![image](https://user-images.githubusercontent.com/29790345/191285800-89ed87a4-c5f7-4683-bc9e-064196bb234f.png)
![image](https://user-images.githubusercontent.com/29790345/191287668-66907e42-bbd1-4381-8e70-b356a3b5cba3.png)
![image](https://user-images.githubusercontent.com/29790345/191297474-f6ee4035-d9ed-4a73-a1e3-5f0ab5227ffe.png)
![image](https://user-images.githubusercontent.com/29790345/191297987-f01a6da0-e07a-4a71-88ef-38d7529dab8d.png)
![image](https://user-images.githubusercontent.com/29790345/191298211-bcdec91e-cff3-4b99-8115-4735049fedb9.png)
![image](https://user-images.githubusercontent.com/29790345/191300086-179ba3fa-8656-42c6-a98c-94a8ea1c7841.png)


### Table View:
![image](https://user-images.githubusercontent.com/29790345/191298747-ce7e411d-3ccc-44c6-9c10-b25ab4c665bd.png)
![image](https://user-images.githubusercontent.com/29790345/191300200-bb98dd87-35b1-42cc-99a6-37c630c84645.png)


### Working Project Snippets:
![image](https://user-images.githubusercontent.com/29790345/191299732-40517772-797b-41aa-9c4e-775292825550.png)
![image](https://user-images.githubusercontent.com/29790345/191301200-80f35587-835c-4f45-98a5-b4d8c1ac9bb1.png)

**Comment by 0xStryK3R** [comment]
### Challenge 21 Tasks:

- [x] Go through the Git Alias documentation shared here
- [x] Create at least 2 alias for any git commands you feel is too long
![image](https://user-images.githubusercontent.com/29790345/191432488-f486ebcb-751c-49e4-80c7-c776db6d801e.png)

- [x] Take a screenshot of the alias you created. hint run git config --list
![image](https://user-images.githubusercontent.com/29790345/191432656-1e8ebd61-fef4-4c62-a2fe-5c4288d4eb78.png)

- [x] Take a screenshot of using your alias
![image](https://user-images.githubusercontent.com/29790345/191433366-9891b04b-7f2a-447c-8691-ee85fb632d19.png)
![image](https://user-images.githubusercontent.com/29790345/191433494-72369a4d-d4fe-4838-b47e-c784c6c2d905.png)

- [x] Post these screenshot in the issue created in the prior challenges

**Comment by kuraakhilesh8230** [comment]
congrats you have completed the milestone-2 successfully and you are good to go🎉

**Comment by 0xStryK3R** [comment]
> congrats you have completed the milestone-2 successfully and you are good to go🎉

Thankyou!! ☺️

**Comment by 0xStryK3R** [comment]
# Challenge 23 Submission:

## [Git Branches](https://codingsight.com/git-branching-naming-convention-best-practices/)

### Branch Categories

- Regular Git Branches:
  - Development (dev)
  - Master (master)
  - QA (QA)

- Temporary Git Branches:
  - Bug Fix
  - Hot Fix
  - Feature Branches
  - Experimental Branches
  - WIP branches

### Git Branching Naming Convention

1. Start branch name with a Group word.
2. Use Unique ID in branch names.
3. Use Hyphen or Slash as Separators.
4. Git Branch with Author Name.
5. Avoid using numbers only.
6. Avoid using all naming convention simultaneously.
7. Avoid long descriptive names for long-lived branches.

## [Pull Requests](https://tighten.com/blog/building-a-great-pull-request/)

### Useful Tips

1. Give your feature branch a clear name.
    > A well-formed PR should start with a well-named feature branch.  

2. Give your commits and PRs active and descriptive titles.  
    > For both commit messages and PR titles, it's a good idea to use the present imperative tense — for example, use “Fix dashboard typo” instead of “Fixed” or “Fixes.”

3. Streamline your process of creating a new PR.  

4. Give your PR a meaningful description
    > A PR description section is where you let the reviewer know why you opened the pull request; the more information you give them before they look at your actual code, the easier their job will be.  
    > A basic description usually has these elements: `Why` `How` and `What`

5. Show your functionality visually, whenever possible

6. Review your own PR before you assign it to others
    >
    > - Make sure you are caught up with the latest changes in the develop branch; resolve any conflicts that might exist, so that merging is as easy as pressing a button.
    > - Run your tests!  
    > - Lint your code!

## [Contributing Guidelines](https://opensource.creativecommons.org/contributing-code/)

### Finding an Issue to Work On:

- Issues available for community contribution:
  - `help wanted`: Open to participation from the community but not necessarily beginner-friendly
  - `good first issue`: Open to participation from the community and friendly towards new contributors
- Issues not available for community contribution:
  - `🔒 staff only`: Requires infrastructure access or institutional knowledge that would be impractical to provide to the community
- Issues not ready for work:
  - `🚧 status`: blocked: Blocked by other work that needs to be done first
  - `🧹 status`: ticket work required: Needs additional work before it is ready to be taken up
  - `🚦 status`: awaiting triage: Has not been triaged by a maintainer
- Issues without any of the above labels.

### Contribution Process

1. Identify issue to work on basis on above guidelines, and comment on it to avoid conflicts.
2. Write code and submit PR. PR should follow [guidleines](https://opensource.creativecommons.org/contributing-code/pr-guidelines/).
3. Wait for code review and address any issues raised as soon as you can.

### Submitting a New Issue

1. Create a new GitHub issue associated with the relevant repository and propose your change there.
2. Wait for a project maintainer to evaluate your issue and decide whether it's something that we will accept a pull request for.
3. Once the project maintainer has approved the issue (by removing the `🚦 status: awaiting triage` label), start work on code as described in above section.

**Comment by 0xStryK3R** [comment]
# Challenge 24 Update:

- [x] I have requested to be added as contributor for the project [deep-translator](https://github.com/nidhaloff/deep-translator). 
- [x] They are looking to build website/api for the deep-translator package. I have [requested](https://github.com/nidhaloff/deep-translator/issues/144#issuecomment-1256920841) to contribute for the same.

**Comment by 0xStryK3R** [comment]
> # Challenge 24 Update:
> * [x]  I have requested to be added as contributor for the project [deep-translator](https://github.com/nidhaloff/deep-translator).
> * [x]  They are looking to build website/api for the deep-translator package. I have [requested](https://github.com/nidhaloff/deep-translator/issues/144#issuecomment-1256920841) to contribute for the same.

# Challenge 25 Update:

- [x] No update received on the previous request submitted yesterday. Awaiting response from owner.
- [x] Also looking for any other Open source repos I can contribute in as per my skillset in case above one doesn't pan out, haven't found anything else yet.

**Comment by 0xStryK3R** [comment]
# Challenge 26 Tasks:
- [x] Do register today for https://hacktoberfest.com/
- [x] Help out other folks, collaborate with each other as well for contributing to open source projects

![IMG_20220927_063006](https://user-images.githubusercontent.com/29790345/192407166-fb4305e7-684d-42e8-825e-a47c672755dc.jpg)

**Comment by 0xStryK3R** [comment]
# Challenge 27 Tasks:
- [x] Go through the rules and resources provided by Hacktoberfest
- [x] Explore different open soure repositories where you might be interested in contributing and look at any potential issues you might want to work on for Hacktoberfest
- [x] Share your tips in the issue created in the previous challenges

# Submission:

___[`🌐 Repos Participating in Hacktoberfest 🌐`](https://github.com/topics/hacktoberfest)___

## Contribution Guidelines:

### Participant Categories: 
- `Contributors`
- `Maintainers`

### Pull/Merge Request Details:
- `OUT-OF-BOUNDS`: PR/MR should be within time bounds (Oct 01 - Oct 31).
- `EXCLUDED`: Repos found violating Hacktoberfest Values will be excluded, and contribution to those will not count.
- `SPAM`: PRs/MRs made shouldn't be spammy.
- `PARTICIPATING`: PRs/MRs made to only participating repos will be considered. Such repo must have topic `HACKTOBERFEST` or PR/MR should be labeled `HACKTOBERFEST-ACCEPTED`.
- `INVALID` : PR/MRS must not be labeled `INVALID`.
- `ACCEPTED` : PR/MRS must be merged, have `HACKTOBERFEST-ACCEPTED` label, or have an overall approving review.

> Once the PR/MRS passes all the above checks, it will be accepted after a 7-Day review period.

### MEASURES TO REDUCE SPAM
- Spammy PRs/MRs will be labelled as `spam`.
- PRs/MRs need to be approved by a `maintainer`.
- Bad repos will be excluded.
- Low Quality PRs/MRs should be avoided.

**Comment by 0xStryK3R** [comment]
> > # Challenge 24 Update:
> > 
> > * [x]   I have requested to be added as contributor for the project [deep-translator](https://github.com/nidhaloff/deep-translator).
> > * [x]   They are looking to build website/api for the deep-translator package. I have [requested](https://github.com/nidhaloff/deep-translator/issues/144#issuecomment-1256920841) to contribute for the same.
> 
> # Challenge 25 Update:
> * [x]  No update received on the previous request submitted yesterday. Awaiting response from owner.
> * [x]  Also looking for any other Open source repos I can contribute in as per my skillset - in case above one doesn't pan out. Haven't found anything else yet.

# Challenge 28 Update:

- [x] No update received on the previous request submitted. Awaiting response from owner.
- [x] Had also submitted a new request on a different [repo](https://github.com/sherlock-project/sherlock/issues/1388) just in case. Owner has assigned me to pick it up, and I have started working on the same.
![image](https://user-images.githubusercontent.com/29790345/192787085-2995cc9b-ad2d-44f9-bfee-6a9a0d4d4ae2.png)

**Comment by 0xStryK3R** [comment]
# Challenge 29 Submission:

- [x] Idea Pitched: [Data Structures And Algorithms Repo](https://github.com/scaleracademy/open-source-projects/discussions/93)
- [x] Screenshot:
![image](https://user-images.githubusercontent.com/29790345/193024854-b6dfafbf-b2d9-4866-8492-992f93d75fd4.png)

**Comment by 0xStryK3R** [comment]
# Challenge 30 Submission:

This was one of the best challenges I have participated in! Learnt a lot about Open Source Contribution, and looking forward to start my own open source journey. Also looking forward for more of the promised events! 😀

**Comment by 0xStryK3R** [comment]
[Milestone 4 Post Link!](https://twitter.com/0xStryK3R/status/1575846485731643393?t=dsLg3OCaje-_2oTjhWUSIw&s=19)

**Comment by kuraakhilesh8230** [comment]
congrats you have completed milestone-4 successfully and you are good to go🎉
